src/txorphanage.h

```diff
@@ -65,6 +70,16 @@ class TxOrphanage {
         LOCK(m_mutex);
         return m_orphans.size();
     }
+    /** Protect an orphan from eviction from the orphanage getting full. The orphan may still be
+     * removed due to expiry. If the orphan is already protected (by any peer), nothing happens.
+     */
+    void ProtectOrphan(const uint256& wtxid) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex);
```

Comment:
Should we implement blessing of all the orphans of a peer with “relay” permissions ? I can definitely see Lightning deployment where you have an inner Bitcoin Core node and multiple external ones, where you load-balance your flows to avoid transaction-relay fingerprinting based on obvious patterns like HTLC-preimage linked to off-chain channel XYZ.